start_admin_sshd.sh

```diff
@@ -47,6 +47,26 @@ get_user_data_keys() {
     ( IFS=$'\n'; echo "${valid_keys[*]}" )
 }
 
+# Export proxy environment variables for all users' login shells
+# Match the values of the proxy environment variables given to the admin container
+set_up_proxy_env_vars() {
+  PROXY_ENV_VAR_SETUP_SCRIPT="/etc/profile.d/br-proxy-env.sh"
+  local export_https_proxy
+  local export_no_proxy
+  if [[ ! -z ${HTTPS_PROXY:+x} ]]; then
+    export_https_proxy="export HTTPS_PROXY=${HTTPS_PROXY} https_proxy=${HTTPS_PROXY}"
+  fi
+  if [[ ! -z ${NO_PROXY:+x} ]]; then
+    export_no_proxy="export NO_PROXY=${NO_PROXY} no_proxy=${NO_PROXY}"
+  fi
+  cat <<EOF > ${PROXY_ENV_VAR_SETUP_SCRIPT}
+# Export Bottlerocket proxy environment variables
+${export_https_proxy}
+${export_no_proxy}
+EOF
```

Comment:
Instead of accumulating the variables with string built commands, this can instead be driven with a "safer" use of `declare -p`:

```bash
cat <<EOF
# Bottlerocket proxy settings
$(declare -p HTTP_PROXY HTTPS_PROXY no_proxy NO_PROXY 2>/dev/null)
EOF
```
<details>
<summary>Example of this used</summary>

```bash
( set -x; export HTTP_PROXY="http://myproxy.example.com/cool/url that's awesome"; export HTTPS_PROXY="https://myproxy.example.com/really/awesome url?ruok=irok"; unset no_proxy; echo "# declare -p ..."; declare -p HTTP_PROXY HTTPS_PROXY http_proxy https_proxy no_proxy; echo "# stderr redirected:"; declare -p HTTP_PROXY HTTPS_PROXY http_proxy https_proxy no_proxy 2>/dev/null; )
```

```bash
# declare -p ...
declare -x HTTP_PROXY="http://myproxy.example.com/cool/url that's awesome"
declare -x HTTPS_PROXY="https://myproxy.example.com/really/awesome url?ruok=irok"
bash: declare: http_proxy: not found
bash: declare: https_proxy: not found
bash: declare: no_proxy: not found
# stderr redirected:
declare -x HTTP_PROXY="http://myproxy.example.com/cool/url that's awesome"
declare -x HTTPS_PROXY="https://myproxy.example.com/really/awesome url?ruok=irok"
```
</details>